tests/integration-tests/tests/networking/test_networking.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import datetime
+import os
+
+import boto3
+import pytest
+
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack, CfnStacksFactory
+from conftest import _AVAILABILITY_ZONE_OVERRIDES
+
+TIMESTAMP = "-{:%Y%m%d%H%M%S}".format(datetime.datetime.utcnow())
+
+
+@pytest.fixture()
+def networking_stack_factory(request):
+    """Define a fixture to manage the creation and destruction of CloudFormation stacks."""
+    factory = CfnStacksFactory()
+    yield factory
+    if not request.config.getoption("no_delete"):
+        factory.delete_all_stacks()
+
+
+def get_internet_gateway_id(ec2_client, vpc_id):
+    internet_gateway_response = ec2_client.describe_internet_gateways(
+        Filters=[{"Name": "attachment.vpc-id", "Values": [vpc_id]}]
+    )
+    return (
+        internet_gateway_response["InternetGateways"][0]["InternetGatewayId"]
+        if internet_gateway_response["InternetGateways"]
+        else ""
+    )
+
+
+def get_cfn_parameters(availability_zone, internet_gateway_id, public_cidr, vpc_id, private_cidr=None):
+    """Create cloudformation-compatible stack parameter given the variables."""
+    parameters = [
+        {"ParameterKey": "AvailabilityZone", "ParameterValue": availability_zone},
+        {"ParameterKey": "InternetGatewayId", "ParameterValue": internet_gateway_id},
+        {"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr},
+        {"ParameterKey": "VpcId", "ParameterValue": vpc_id},
+    ]
+    if private_cidr:
+        parameters.append({"ParameterKey": "PrivateCIDR", "ParameterValue": private_cidr})
+    return parameters
+
+
+def load_public_template():
+    with open(os.path.join("..", "..", "cloudformation", "networking_configuration", "public.cfn.json")) as cfn_file:
+        file_content = cfn_file.read()
+    return file_content
+
+
+def load_public_private_template():
+    with open(
+        os.path.join("..", "..", "cloudformation", "networking_configuration", "public-private.cfn.json")
+    ) as cfn_file:
+        file_content = cfn_file.read()
+    return file_content
+
+
+def test_public_configuration_no_internet_gateway(region, vpc_stacks, networking_stack_factory):
```

Comment:
minor: I would chose a slightly better name for the test such as:
test_public_network_topology/configuration

without mentioning the no_internet_gateway cause otherwise it seems the network does not have an IGW at all.

Reply:
minor you could have a fixture vpc_stack(vpc_stacks, region) that returns only the vpc_stack for the specific region under test